### PR TITLE
storage: add TestWorkflow

### PR DESCRIPTION
**Pull request by yichengq**
The first PR fixes a bug that happens in this test:
```
2015/06/17 13:21:25 storage: restore compact to 0
2015/06/17 13:21:25 storage: cannot unmarshal event finishedCompactRev: proto: illegal wireType 7
exit status 1
```

The second PR adds a workflow test that simulates the workflow that storage is used in normal etcd running, including key changes, compaction and restart.

### CODE REVIEW

File: storage/kvstore.go [hunk 1]
```diff
@@ -14,9 +14,10 @@ import (
 )
 
 var (
-	batchLimit    = 10000
-	batchInterval = 100 * time.Millisecond
-	keyBucketName = []byte("key")
+	batchLimit     = 10000
+	batchInterval  = 100 * time.Millisecond
+	keyBucketName  = []byte("key")
+	infoBucketName = []byte("information")
```
Comment (xiang90) [inline storage/kvstore.go]: how about changing this to metaBucketName?

Comment (yichengq) [inline storage/kvstore.go]: sure. and "meta" for table name then.

**Comment by yichengq** [comment]
addressed all comments.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**